### DISCUSSION TITLE
[Feature] settings import/export

### DISCUSSION THREAD

**Discussion by dpantel** [linuxserver/Heimdall#692] [Ideas]
Feature request:

I would love an easy way to export and import the settings.  It would help greatly with automated deployment (Ansible, etc.) of Heimdall.

I know I can copy the sqlite db, but that seems messy and might interfere with db changes in the future.

Some ideas:
1. The most rudimentary way to implement this would be a simple sql dump and ingest.  This way, all you need is a text file.
    - upsides
      - easy to implement
    - downsides
      - hard to read/edit manually
      - have to trust that text file is well-formed

1. A more complicated, but better way would be to transform the necessary data to/from a config file.
    - downsides
		- harder to implement
	- upsides
		- easy to read/edit
		- well-structured and well-documented syntax
		- there are [parsers/validators/builders](https://laravel-news.com/toml-configuration-in-php) already available in Laravel
    - formats to consider
    	- TOML
    	- YAML
    	- JSON
    	- INI

**Comment by DavidRawling** [comment]
An alternative or extension to this might be for the admin to be able to add the app to a "shared" section of the heimdall world, which then makes it available to all users for pinning (but not customisation).

**Comment by dakipro** [comment]
I would also like this as I would like to provision server and include heimdall dashboard preconfigured based on services server is heaving. Perhaps even a web API would work.

**Comment by justjake** [comment]
I just finished setting up a new server. The hiemdall part is the only thing that I haven't figured out how to automate :(.

**Comment by justjake** [comment]
See also #383 ”Web GUI Backup”.

**Comment by makcuber** [comment]
Just got Heimdall running in docker to test it out. The lack of some sort of bulk import feature is a big negative for me. I have a ton of URL's preorganized as bookmarks, and I'd love to be able to just export those and import them into Heimdall so they are accessible as a slashpage
